src/QirRuntime/lib/Simulators/ToffoliSimulator.cpp

```diff
@@ -67,21 +65,23 @@ namespace Quantum
         Qubit AllocateQubit() override
         {
             this->lastUsedId++;
-            this->states.ExtendToInclude(this->lastUsedId);
+            this->states.emplace_back();
             return reinterpret_cast<Qubit>(this->lastUsedId);
         }
 
         void ReleaseQubit(Qubit qubit) override
         {
             const long id = GetQubitId(qubit);
             assert(id <= this->lastUsedId);
-            assert(!this->states.IsBitSetAt(id));
+            assert(!this->states.at(id));
+            this->lastUsedId--;
+            this->states.pop_back();

```

Comment:
Note that this does have a change in behavior here, where previously an allocated qubit that was released would never be used again, here we allow for qubit "reuse" by decrementing the `lastUsedId` member such that future allocations can use that id again. This will prevent the bit vector from always growing, such that Q# like this:
```qsharp
for i in 1..10 {
    use q = Qubit();
    X(q);
    Reset(q);
}
```
Will now have a bit vector of at most length 1 instead of length 10.